src/map/status.c

```diff
@@ -13397,25 +13397,102 @@ static bool status_readdb_sizefix(char *fields[], int columns, int current)
 	return true;
 }
 
-static bool status_readdb_scconfig(char *fields[], int columns, int current)
+static bool status_read_scdb_libconfig(void)
 {
-	int val = 0;
-	char* type = fields[0];
+	struct config_t status_conf;
+	char filepath[256];
+	safesnprintf(filepath, sizeof(filepath), "%s/%s", map->db_path, "sc_config.conf");
 
-	nullpo_retr(false, fields);
-	if( !script->get_constant(type, &val) ){
-		ShowWarning("status_readdb_sc_conf: Invalid status type %s specified.\n", type);
+	if (libconfig->load_file(&status_conf, filepath) == CONFIG_FALSE) {
+		ShowError("status_read_scdb_libconfig: can't read %s\n", filepath);
 		return false;
 	}
 
-	status->dbs->sc_conf[val] = (int)strtol(fields[1], NULL, 0);
-	if (status->dbs->sc_conf[val] & SC_VISIBLE)
-	{
-		status->dbs->DisplayType[val] = true;
+	int i = 0;
+	int count = 0;
+	struct config_setting_t *it = NULL;
+
+	while ((it = libconfig->setting_get_elem(status_conf.root, i++)) != NULL) {
+		if (status->read_scdb_libconfig_sub(it, i - 1, filepath))
+			++count;
+	}
+
+	libconfig->destroy(&status_conf);
+	ShowStatus("Done reading '"CL_WHITE"%d"CL_RESET"' entries in '"CL_WHITE"%s"CL_RESET"'.\n", count, filepath);
+	return true;
+}
+
+static bool status_read_scdb_libconfig_sub(struct config_setting_t *it, int idx, const char *source)
+{
+	nullpo_retr(false, it);
+	nullpo_retr(false, source);
+
+	int i32;
+	const char *name = config_setting_name(it);
+
+	if (!script->get_constant(name, &i32) || i32 <= SC_NONE || i32 >= SC_MAX) {
+		ShowWarning("status_read_scdb_libconfig_sub: Invalid status type (%s) in \"%s\" entry #%d, skipping.\n", name, source, idx);
+		return false;
 	}
 
+	libconfig->setting_lookup_bool_real(it, "Visible", &status->dbs->DisplayType[i32]);
+
+	struct config_setting_t *fg = libconfig->setting_get_member(it, "Flags");
```

Comment:
the old format allowed flags to be optional, aka value of 0, perhaps we should retrain this in the new feature, right bow it will assert.